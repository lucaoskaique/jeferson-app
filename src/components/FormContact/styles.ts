import * as ButtonStyles from 'components/Button/styles'
import * as TextAreaStyles from 'components/TextArea/styles'
import * as TextFieldStyles from 'components/TextField/styles'
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const BoxEmailName = styled.div`
  ${media.greaterThan('medium')`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  `}
`
export const BoxCityPhone = styled.div`
  ${media.greaterThan('medium')`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  `}
`
export const BoxButton = styled.div`
  text-align: right;
`
export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${TextFieldStyles.Wrapper},
    ${TextAreaStyles.Wrapper} {
      margin: ${theme.spacings.xxsmall} 0;
    }

    ${ButtonStyles.Wrapper} {
      margin: ${theme.spacings.medium} auto ${theme.spacings.xsmall};
    }

    ${media.greaterThan('medium')`
      ${TextFieldStyles.Wrapper},
      ${TextAreaStyles.Wrapper} {
        margin: ${theme.spacings.xxsmall} ${theme.spacings.xxsmall};
      }
    `}
  `}
`
