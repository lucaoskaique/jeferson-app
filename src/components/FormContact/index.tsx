import {
  Email,
  LocationCity,
  Person,
  Phone
} from '@styled-icons/material-outlined'
import Button from 'components/Button'
import TextArea from 'components/TextArea'
import TextField from 'components/TextField'

import * as S from './styles'

const FormContact = () => (
  <S.Wrapper>
    <form>
      <S.BoxEmailName>
        <TextField
          name="email"
          placeholder="E-mail"
          type="email"
          icon={<Email />}
          pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$"
        />
        <TextField
          name="name"
          type="text"
          placeholder="Nome e sobrenome"
          icon={<Person />}
          pattern="[a-z\s]+$"
        />
      </S.BoxEmailName>
      <S.BoxCityPhone>
        <TextField
          name="city"
          type="text"
          placeholder="Cidade"
          icon={<LocationCity />}
          pattern="[a-z\s]+$"
        />
        <TextField
          name="phone"
          type="tel"
          placeholder="(99)99999-9999"
          icon={<Phone />}
          pattern="\([0-9]{2}\) [0-9]{4,6}-[0-9]{3,4}$"
          maxLength={15}
        />
      </S.BoxCityPhone>

      <TextArea name="message" placeholder="Sua mensagem aqui" />
      <S.BoxButton>
        <Button size="small">Enviar</Button>
      </S.BoxButton>
    </form>
  </S.Wrapper>
)

export default FormContact
