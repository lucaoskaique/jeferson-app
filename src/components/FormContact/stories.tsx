import { Story, Meta } from '@storybook/react'

import FormContact from '.'

export default {
  title: 'Form/FormContact',
  component: FormContact
} as Meta

export const Default: Story = () => (
  <div style={{ width: 300, margin: 'auto' }}>
    <FormContact />
  </div>
)
