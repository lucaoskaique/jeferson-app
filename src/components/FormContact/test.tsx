import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormContact from '.'

describe('<FormContact />', () => {
  it('should render the form', () => {
    const { container } = renderWithTheme(<FormContact />)

    expect(screen.getByPlaceholderText(/E-mail/i)).toBeInTheDocument()

    expect(screen.getByPlaceholderText(/Nome e sobrenome/i)).toBeInTheDocument()

    expect(screen.getByPlaceholderText(/Cidade/i)).toBeInTheDocument()

    expect(screen.getByPlaceholderText('(99)99999-9999')).toBeInTheDocument()

    expect(
      screen.getByPlaceholderText(/Sua mensagem aqui/i)
    ).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /enviar/i })).toBeInTheDocument()
    expect(container.parentElement).toMatchSnapshot()
  })
})
