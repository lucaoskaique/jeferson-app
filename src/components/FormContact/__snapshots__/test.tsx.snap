// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormContact /> should render the form 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  height: 3rem;
  font-size: 1.2rem;
  background: linear-gradient(90deg,#ffbf3a -40%,#d90726 185.31%);
}

.c13:focus {
  outline: 1px dashed;
}

.c13:hover {
  background: linear-gradient(90deg,#72c042 -99.98%,#0092d3 193.73%);
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FAFAFA;
  border-radius: 1rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c9:focus-within {
  box-shadow: 0 0 0.5rem #EC6330;
}

.c10 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c10:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c10:-webkit-autofill::first-line {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FAFAFA;
  border-radius: 1rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c3:focus-within {
  box-shadow: 0 0 0.5rem #EC6330;
}

.c6 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c6:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c6:-webkit-autofill::first-line {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #8F8F8F;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c4 > svg {
  width: 2.2rem;
  height: 100%;
}

.c11 {
  text-align: right;
}

.c0 .c2,
.c0 .c8 {
  margin: 0.8rem 0;
}

.c0 .c12 {
  margin: 3.2rem auto 1.6rem;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c7 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c0 .c2,
  .c0 .c8 {
    margin: 0.8rem 0.8rem;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <form>
        <div
          class="c1"
        >
          <div
            class="c2 "
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <svg
                  aria-hidden="true"
                  class="c5"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
                  />
                </svg>
              </div>
              <input
                class="c6"
                name="email"
                pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
                placeholder="E-mail"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="c2 "
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <svg
                  aria-hidden="true"
                  class="c5"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 10c2.7 0 5.8 1.29 6 2H6c.23-.72 3.31-2 6-2m0-12C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 10c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <input
                class="c6"
                name="name"
                pattern="[a-z\\\\s]+$"
                placeholder="Nome e sobrenome"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="c7"
        >
          <div
            class="c2 "
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <svg
                  aria-hidden="true"
                  class="c5"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
                  />
                </svg>
              </div>
              <input
                class="c6"
                name="city"
                pattern="[a-z\\\\s]+$"
                placeholder="Cidade"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="c2 "
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <svg
                  aria-hidden="true"
                  class="c5"
                  fill="currentColor"
                  focusable="false"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6.54 5c.06.89.21 1.76.45 2.59l-1.2 1.2c-.41-1.2-.67-2.47-.76-3.79h1.51m9.86 12.02c.85.24 1.72.39 2.6.45v1.49c-1.32-.09-2.59-.35-3.8-.75l1.2-1.19M7.5 3H4c-.55 0-1 .45-1 1 0 9.39 7.61 17 17 17 .55 0 1-.45 1-1v-3.49c0-.55-.45-1-1-1-1.24 0-2.45-.2-3.57-.57a.84.84 0 00-.31-.05c-.26 0-.51.1-.71.29l-2.2 2.2a15.149 15.149 0 01-6.59-6.59l2.2-2.2c.28-.28.36-.67.25-1.02A11.36 11.36 0 018.5 4c0-.55-.45-1-1-1z"
                  />
                </svg>
              </div>
              <input
                class="c6"
                maxlength="15"
                name="phone"
                pattern="\\\\([0-9]{2}\\\\) [0-9]{4,6}-[0-9]{3,4}$"
                placeholder="(99)99999-9999"
                type="tel"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="c8 "
        >
          <div
            class="c9"
          >
            <textarea
              class="c10"
              name="message"
              placeholder="Sua mensagem aqui"
            />
          </div>
        </div>
        <div
          class="c11"
        >
          <button
            class="c12 c13"
          >
            <span>
              Enviar
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`
